### PR TITLE
windows: Fix extensions couldn't start if the path contained spaces

### LINKED ISSUE #15441: Tailwind CSS language server doesn't escape space in path.
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

My username in Windows is Aldo Ridhoni. When opening a project with tailwindcss-language-enabled, a windows popup titled "Select an App to Open Aldo" is shown.

### Environment

Zed 0.147.0
Windows 11

### If applicable, add mockups / screenshots to help explain present your vision of the feature

Icannot screenshot the Select an App to Open Aldo popup. 
Here is screenshot of powershell event log. 
image

### If applicable, attach your Zed.log file to this issue.

<details><summary>Zed.log</summary><pre>

Caused by:
    The system cannot find the file specified. (os error 2)
2024-07-29T22:37:38.6097415+07:00 [INFO] starting language server "tailwindcss-language-server", path: "C:\\Users\\Aldo Ridhoni\\XXXX\\XXXX\\XXXX", id: 7
2024-07-29T22:37:38.610115+07:00 [INFO] starting language server. binary path: "powershell.exe", working directory: "C:\\Users\\Aldo Ridhoni\\XX\\XX\\XXX", args: ["C:\\Users\\Aldo Ridhoni\\AppData\\Local\\Zed\\languages\\tailwindcss-language-server\\node_modules/.bin/tailwindcss-language-server.ps1", "--stdio"]
2024-07-29T22:37:38.8569836+07:00 [ERROR] cannot read LSP message headers
2024-07-29T22:37:38.8572663+07:00 [ERROR] failed to start language server "tailwindcss-language-server": oneshot canceled
2024-07-29T22:37:38.8574055+07:00 [ERROR] server stderr: Some("")
2024-07-29T22:37:38.8575424+07:00 [INFO] retrying installation of language server "tailwindcss-language-server" in 1s
2024-07-29T22:37:38.8577201+07:00 [ERROR] server shut down
2024-07-29T22:37:39.861547+07:00 [ERROR] The system cannot find the path specified. (os error 3)
2024-07-29T22:37:39.8620616+07:00 [INFO] About to spawn test binary
2024-07-29T22:37:39.8624025+07:00 [WARN] test binary failed to launch
2024-07-29T22:37:39.8627131+07:00 [WARN] test binary check failed
2024-07-29T22:37:39.8630149+07:00 [INFO] beginning to reinstall server
2024-07-29T22:37:39.8633677+07:00 [INFO] deleting server container
2024-07-29T22:37:39.8640798+07:00 [ERROR] server container removal
</pre></details>

### PR DESCRIPTION
**Pull request by JunkuiZhang**
Closes #15441 .

Fixed the issue where extensions couldn't start if the path contained spaces. Additionally, this PR introduces the `node_environment_path` function to obtain the PATH environment variable which includes the node path.

Release Notes:

- N/A

### CODE REVIEW

File: crates/node_runtime/src/node_runtime.rs [hunk 1]
```diff
@@ -216,6 +218,22 @@ impl NodeRuntime for RealNodeRuntime {
         Ok(installation_path.join(NODE_PATH))
     }
 
+    async fn node_environment_path(&self) -> Result<OsString> {
```
Comment (JunkuiZhang) [inline crates/node_runtime/src/node_runtime.rs:221]: And any idea on this function name?

Comment (huacnlee) [reply crates/node_runtime/src/node_runtime.rs:221]: `node_path` or `node_home` ? 

Like `GOPATH` env in Go.

Comment (JunkuiZhang) [reply crates/node_runtime/src/node_runtime.rs:221]: > GOPATH variable (which defaults to $HOME/go on Unix and %USERPROFILE%\go on Windows) is used for the following purposes
This is a little different from `GOPATH`; here we are retrieving the `PATH` environment variable that contains the node path.

Comment (mikayla-maki) [reply crates/node_runtime/src/node_runtime.rs:221]: I think that seems like a good name.

**Comment by vipexv** [comment]
Sorry to bother but the Tailwind CSS Server still doesn't work for me.

Zed Logs: 


https://gist.github.com/vipexv/ef62946efb9855644e254e18a5a24511

**Comment by JunkuiZhang** [comment]
> Sorry to bother but the Tailwind CSS Server still doesn't work for me.
> 
> Zed Logs:
> 
> https://gist.github.com/vipexv/ef62946efb9855644e254e18a5a24511

It complains that can not find the target file. Dose deleting  the folder `C:\\Users\\v\\AppData\\Local\\Zed\\languages\\tailwindcss-language-server` working for you?

**Comment by vipexv** [comment]
I don't even have that folder tho.

**Pull request merged**